core/app/components/pubs/PubEditor/actions.ts

```diff
@@ -45,24 +45,26 @@ export const createPub = defineServerAction(async function createPub({
 	const pubValueEntries = Object.entries(pubValues);
 
 	try {
-		const query = db
-			.with("new_pub", (db) =>
-				db
-					.insertInto("pubs")
-					.values({
-						id: pubId,
-						communityId: communityId,
-						pubTypeId: pubTypeId,
-						parentId: parentId,
-					})
-					.returning("id")
-			)
-			.with("stage_create", (db) =>
-				db.insertInto("PubsInStages").values((eb) => ({
-					pubId: eb.selectFrom("new_pub").select("new_pub.id"),
-					stageId,
-				}))
-			);
+		const pubQuery = db.with("new_pub", (db) =>
+			db
+				.insertInto("pubs")
+				.values({
+					id: pubId,
+					communityId: communityId,
+					pubTypeId: pubTypeId,
+					parentId: parentId,
+				})
+				.returning("id")
+		);
+
+		const query = stageId
+			? pubQuery.with("stage_create", (db) =>
+					db.insertInto("PubsInStages").values((eb) => ({
+						pubId: eb.selectFrom("new_pub").select("new_pub.id"),
+						stageId,
+					}))
+				)
+			: pubQuery;
```

Comment:
kind of ugly, but this fixes the type issue below.

For readability's sake though, it might be worth splitting these paths up into separate functions anyway though. Even if it causes some code duplication.

```suggestion
		const query = (
			stageId
				? pubQuery.with("stage_create", (db) =>
						db.insertInto("PubsInStages").values((eb) => ({
							pubId: eb.selectFrom("new_pub").select("new_pub.id"),
							stageId,
						}))
					)
				: pubQuery
		) as typeof pubQuery;
```